src/gt4py/backend/dawn_backends.py

```diff
@@ -0,0 +1,595 @@
+            )
+        ]
+
+        return cls.build_extension_module(
+            stencil_id,
+            gt_pyext_sources,
+            pyext_opts,
+            pyext_extra_include_dirs=include_dirs,
+            uses_cuda=uses_cuda,
+        )
+
+    @classmethod
+    @abc.abstractmethod
+    def generate_extension(
+        cls,
+        stencil_id: gt_definitions.StencilID,
+        definition_ir: gt_ir.StencilDefinition,
+        options: gt_definitions.BuildOptions,
+        **kwargs,
+    ):
+        pass
+
+
+_DAWN_BASE_OPTIONS = {
+    "add_profile_info": {"versioning": True},
+    "clean": {"versioning": False},
+    "debug_mode": {"versioning": True},
+    "dump_sir": {"versioning": False},
+    "verbose": {"versioning": False},
+}
+
+
+_DAWN_TOOLCHAIN_OPTIONS = {}
+for name in dir(dawn4py.Options):
+    if (
+        name.startswith("print")
+        or name.startswith("dump")
+        or name.startswith("report")
+        or name.startswith("serialize")
+        or name.startswith("deserialize")
+    ):
+        _DAWN_TOOLCHAIN_OPTIONS[name] = {"versioning": False}
+    elif not name.startswith("_") and name != "backend":
+        _DAWN_TOOLCHAIN_OPTIONS[name] = {"versioning": True}
+
+
+_DAWN_BACKEND_OPTIONS = {**_DAWN_BASE_OPTIONS, **_DAWN_TOOLCHAIN_OPTIONS}
+
+
+@gt_backend.register
+class DawnGTX86Backend(BaseDawnBackend):
+
+    DAWN_BACKEND_NS = "gt"
+    DAWN_BACKEND_NAME = "gridtools"
+    GT_BACKEND_T = "x86"
+
+    name = "dawn:gtx86"
+    options = _DAWN_BACKEND_OPTIONS
+    storage_info = gt_backend.GTX86Backend.storage_info
+
+    @classmethod
+    def generate_extension(cls, stencil_id, definition_ir, options, **kwargs):
+        return cls._generic_generate_extension(
+            stencil_id, definition_ir, options, uses_cuda=False, **kwargs
+        )
+
+
+@gt_backend.register
+class DawnGTMCBackend(BaseDawnBackend):
+
+    DAWN_BACKEND_NS = "gt"
+    DAWN_BACKEND_NAME = "gridtools"
+    GT_BACKEND_T = "mc"
+
+    name = "dawn:gtmc"
+    options = _DAWN_BACKEND_OPTIONS
+    storage_info = gt_backend.GTMCBackend.storage_info
+
+    @classmethod
+    def generate_extension(cls, stencil_id, definition_ir, options, **kwargs):
+        return cls._generic_generate_extension(
+            stencil_id, definition_ir, options, uses_cuda=False, **kwargs
+        )
+
+
+@gt_backend.register
+class DawnGTCUDABackend(BaseDawnBackend):
+
+    DAWN_BACKEND_NS = "gt"
+    DAWN_BACKEND_NAME = "gridtools"
+    GT_BACKEND_T = "cuda"
+
+    MODULE_GENERATOR_CLASS = gt_backend.CUDAPyExtModuleGenerator
+
+    name = "dawn:gtcuda"
+    options = _DAWN_BACKEND_OPTIONS
+    storage_info = gt_backend.GTCUDABackend.storage_info
+
+    @classmethod
+    def generate_extension(cls, stencil_id, definition_ir, options, **kwargs):
+        return cls._generic_generate_extension(
+            stencil_id, definition_ir, options, uses_cuda=True, **kwargs
+        )
+
+
+# @gt_backend.register
```

Comment:
Why is this commented?

Reply:
Naive backend is not working. Vulcan people will likely provide the fixes